src/lazy/text/buffer.rs

```diff
@@ -878,64 +947,76 @@ impl<'data> TextBufferView<'data> {
         // Scan ahead to find the end of this list.
         let list_body = self.slice_to_end(1);
         let sequence_iter = RawTextListIterator_1_1::new(list_body);
-        let span = match sequence_iter.find_span() {
-            Ok(span) => span,
-            // If the complete container isn't available, return an incomplete.
-            Err(IonError::Incomplete(_)) => return Err(nom::Err::Incomplete(Needed::Unknown)),
-            // If invalid syntax was encountered, return a failure to prevent nom from trying
-            // other parser kinds.
-            Err(e) => {
-                return {
-                    let error = InvalidInputError::new(self)
-                        .with_label("matching a v1.1 list")
-                        .with_description(format!("{}", e));
-                    Err(nom::Err::Failure(IonParseError::Invalid(error)))
+        let (span, child_exprs) =
+            match TextListSpanFinder_1_1::new(self.allocator, sequence_iter).find_span() {
```

Comment:
🗺️ The `_______SpanFinder` types wrap a parsing iterator and return the matched span along with any nested expressions.